### ISSUE TITLE
Different results between competition's page and person's page

### ISSUE THREAD

**Issue by viroulep** [cubing/worldcubeassociation.org#772]
Right now when looking at [Saint Mary Scramble 2016](https://www.worldcubeassociation.org/competitions/SaintMaryScramble2016) page the results are not displayed, but if you look at one of the competitors' page (eg: [Chris Olson](https://www.worldcubeassociation.org/results/p.php?i=2009OLSO01)) you can see he already has results for the competition.

Not sure what is going on, most likely results are in the db but somehow not validated? In this case they should probably not appear on the person's page until they are "fully" up.

**Comment by jonatanklosko** [comment]
I can see the results now. I remember there's a moment when results are on the page but are not visible to the public, but I've no idea how PHP code behaves in this case.

**Comment by jfly** [comment]
Our rails code is aware of this new [`results_posted_at`](https://github.com/cubing/worldcubeassociation.org/blob/bad53e5988e752f8a08722ef915ddc552da1619f/WcaOnRails/app/views/competitions/_index_table.html.erb#L10-L13) field, but PHP doesn't know anything about it. It's easy to hide an entire competition, but keeping track of old world rankings vs new world rankings would get tricky if we wanted to handle this perfectly. @larspetrus actually started a discussion with the results team about this back in may: https://groups.google.com/d/msg/wca-admin/hUV7TjToPQM/LI6vCDwPAQAJ.

**Comment by ronaldmansveld** [comment]
@viroulep Is this issue still relevant as both pages have moved to Rails now?

**Comment by viroulep** [comment]
Yes it is, the [current person page](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/views/persons/_results_by_event.html.erb#L27) uses `@results` defined [here](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/controllers/persons_controller.rb#L36), which does not do any check whether these results comes from posted competition or not.

Whereas the competition's page checks that the [user can view results](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/views/competitions/_results_nav.html.erb#L4), which is defined [here](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/models/competition.rb#L754) and checks that the results are posted.
